### DISCUSSION TITLE
FOV issue in CT2D_fanbeam

### DISCUSSION THREAD

**Discussion by omrghad** [villekf/OMEGA#50] [General]
Hi all, 
i hope you are doing well,
i have the issue of field of view, when i raise the FOV it doesn't work, or if the code run successfully, th image is the same, 
this is my code details:

i used "CT2D_fanbeam_mainExample.m"
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Binning
% The level of binning used for the raw data. For example binning of 2
% reduces the size of the projections by two from both dimensions (e.g.
% 2048x3072 becomes 1024x1536).
options.binning = 1;

%%% Number of detector pixels (row)
% The number of detector pixels in the detector panel (row
% direction)
% NOTE: if you use binning, this value has to use the final binned
% dimensions
% In the 2D case, this should be 1
options.nRowsD = 200;

%%% Number of detector pixels (column)
% The number of detector pixels in the detector panel (column
% direction)
% NOTE: if you use binning, this value has to use the final binned
% dimensions
options.nColsD = 1;

%%% Number of projections
% Total number of projections used
options.nProjections = 90;

%%% Projection angles (degree or radian)
% The angles corresponding to the projections
options.angles = -linspace(0, 356, options.nProjections);

%%% Detector pixel pitch/size (mm)
% The size of the detector/distance between adjacent detectors
% NOTE: if you use binning, this value has to use the final binned
% dimensions
options.dPitch = 0.7;

%%% Source to detector distance (mm)
% The orthogonal distance from the source to the detector panel
options.sourceToDetector = 900;

%%% Source to center of rotation distance (mm)
% The distance from the source to the center of rotation/object/origin
options.sourceToCRot = 316;

%%% Name of current datafile/examination
% This is used for naming purposes only
[options.name](http://options.name/) = '2DCT_data';

%%% Show status messages
% These are e.g. time elapsed on various functions and what steps have been
% completed. It is recommended to keep this true.
options.verbose = 1;

%%% Transaxial FOV size (mm), this is the length of the x (horizontal) side
% of the FOV
options.FOVa_x = 350;

%%% Transaxial FOV size (mm), this is the length of the y (vertical) side
% of the FOv
options.FOVa_y = options.FOVa_x;

%%% Axial FOV (mm)
options.axial_fov = options.dPitch;

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% Load the measurement sinogram
% If you use different resolution, be sure the modify the scanner
% properties accordingly
load('/home/hassan/Desktop/output/Data164.mat','m')
 m(m < 0) = 1;
 
 options.SinM = m;
options.SinM = single(options.SinM) ./ single(max(max(max(options.SinM(:)))));
options.SinM(options.SinM > 1) = single(1);
options.SinM(options.SinM == 0) = single(min(options.SinM(options.SinM > 0)));
m=options.SinM;
 
options.SinM = 1./exp(m);
m= options.SinM;

% This is required when the input data is already linearized
options.usingLinearizedData = true;

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%% IMAGE PROPERTIES %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% Reconstructed image pixel size (X-direction)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Reconstructed image pixel size (X-direction)

options.Nx = 512;

%%% Y-direction
options.Ny = 512;

%%% Z-direction (number of slices) (axial)
options.Nz = 1;

could you please help me to resolve this issue?

i tried to use this relation to raise the FOV: FOV=Pixel Size×Matrix Size
, but the code doesn't run and in the end matlab exit.
 the ct slice obtained for FOV of 140.1 mm attached below
thank you very much
best regards

**Comment by villekf** [comment]
As a quick workaround, you can delete `2DCT_data_lor_pixel_count...` files from the mat-files folder. Any "...lor_pixel_count..." files are safe to delete. 

The reason for the crash is that I use previously computed data if possible, but that doesn't take into account changes to the FOV size, only in the number of voxels. Note that this applies to implementation 1 only.

**Comment by omrghad** [comment]
after your solution, the code runs without issue despite i am using implementation 4, thank you for your reply 
but the issue is the same, Please take a look at the attached image, regards
![fov](https://github.com/user-attachments/assets/90f8c3e6-e52c-43bd-94d4-cef3873f8b60)

**Reply by villekf** [reply]
Implementation 4 has a separate issue. Since you are using 2D data, you can use implementation 1 in the meantime. Everything should work fine there as long as you delete the file I mentioned if you change the FOV again. I'll try to fix implementation 4 soon.

**Reply by villekf** [reply]
Implementation 4 should now work in the current master and the FOV can be changed there without any issues. You might want to download the current master, rebuild the mex-files and then try again.

**Reply by omrghad** [reply]
thank you for your updates, but the circle with the red arrow diameter does not changed, does it depending on the detector length ? regards

**Reply by villekf** [reply]
Sorry, I believe I misunderstood your question before. In order to increase diameter of the region where the red arrow is, you also need to increase the size of the detector, i.e. you need more detector elements, either actual ones or extrapolated/estimated ones.

OMEGA has support for extrapolating projections, but I don't think it works with 2D data.

**Reply by omrghad** [reply]
thank you, I'll try to extrapolate if it works, otherwise I'll do simulations with longer detectors, regards

**Comment by omrghad** [comment]
dear villekf, 
as you told me yesterday, i applied a manually extrapolation to the sinogram matrix, it appears that works well, 
the two attached extrapolation ( of 20 pixels and of 100 pixels) attached below,
![extensionof20pixelonsinogrammatrix](https://github.com/user-attachments/assets/23981a1c-59f9-4089-aff9-580dea17d65c)
![extrapolate](https://github.com/user-attachments/assets/2dca800a-8feb-461d-a45e-39fcea492e88)
Kindly,  the area indicated by the red arrow corresponds to what ?
regards

**Reply by villekf** [reply]
If I understood your question correctly, the red area is the region where the fan beam coverage is less than elsewhere. If you visualize the fan beam, you'll see that you have a circular region which has a full coverage from all angles and then the outside region (as indicated by the red arrow) that has less coverage from the fan, i.e. only some of the angles cover a specific region.

**Reply by omrghad** [reply]
i got it, thank you very much
